### PR TITLE
intel/2018a toolchain (+ HPL easyconfig as test case)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (4 easyconfigs in this PR)
node2481.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b844bdf347822e589be05c7ebb0c5cc0 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2057.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/b92a5993b61e570c55ca8d449fae8ce4 for a full test report.

File: easybuild/easyconfigs/h/HPL/HPL-2.2-intel-2018a.eb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ arithmetic on distributed-memory computers. It can thus be regarded as a portable as well as freely available
+ implementation of the High Performance Computing Linpack Benchmark."""
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+toolchainopts = {'usempi': True}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = ['http://www.netlib.org/benchmark/%(namelower)s']
+checksums = ['ac7534163a09e21a5fa763e4e16dfc119bc84043f6e6a807aba666518f8df440']
+
```
Comment (vanzod) [inline easybuild/easyconfigs/h/HPL/HPL-2.2-intel-2018a.eb]: Missing patch checksum

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.9, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.6.6
See https://gist.github.com/08e49f8aae0395fa0fdb2b84d85747ab for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.3, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/08455d313b395840e6f0c89a70f72728 for a full test report.

**Comment by boegel** [comment]
Results from testing existing easyconfigs that use the `intel/2017b` toolchain with `intel/2018a` (210 easyconfigs at the time): 

* HDF5 1.10.1 requires a patch to build correctly
  * was already made available by @reedts via #5310 
  * easyconfig for `intel/2018a` in #5582
* LLVM 5.0.x fails when compiling `llvm-5.0.0.src/lib/IR/Attributes.cpp` with `error: expression must have a constant value`
  * fixed with patch in #5631
* ImageMagick 7.0.7-14 fails with:
  ```
  ": internal error: ** The compiler has encountered an unexpected problem.
  ** Segmentation violation signal raised. **
  Access violation or stack overflow. Please contact Intel Support for assistance.

  compilation aborted for coders/cip.c (code 4)
  ```
  I'll look into reporting this to Intel.

* failures because of hardcoded use of `-openmp` (which is no longer valid: `icpc: command line error: option '-openmp' is not supported. Please use the replacement option '-qopenmp'`);
  * BLAST+ 2.6.0 + 2.7.1
  * ESMF 7.0.2 (+ NCL 6.4.0 that depends on ESMF)
  * to be fixed in upcoming PRs
* other minor issues (will be handled when need arises):
  * DFTB+ 17.1: failing tests
  * mkl-dnn 0.11: compilation fails because a loop was not vectorised (?!)
  * MIRA 4.0.2: `error: no instance of overloaded function "MIRAParameters::parse" matches the argument list`

**Pull request merged**